### PR TITLE
2299 remove op rep

### PR DESCRIPTION
**Pull request by BCerki**
card: https://github.com/orgs/bcgov/projects/123/views/16?pane=issue&itemId=81400225&issue=bcgov%7Ccas-registration%7C2299

This PR:
- new widget for operation reps
- vitests^
- new endpoint to remove a rep
- pytests^

I wasn't sure what to return from the endpoint since we don't need anything. I tried to return nothing (suggestions for something better?), but I'm getting {id: None}/{id: null} for some reason.

### CODE REVIEW

File: bc_obps/registration/tests/endpoints/v2/_operations/_operation_id/_registration/test_operation_representative.py [hunk 1]
```diff
@@ -171,3 +171,73 @@ def test_register_operation_operation_representative_endpoint_bad_data(self):
...
+            'id': contact.id,
+        }
+        response = TestUtils.mock_put_with_auth_role(
+            self,
+            "industry_user",
+            self.content_type,
+            data,
+            custom_reverse_lazy("remove_operation_representative", kwargs={'operation_id': operation.id}),
+        )
+        assert response.status_code == 200
```
Comment (BCerki) [inline bc_obps/registration/tests/endpoints/v2/_operations/_operation_id/_registration/test_operation_representative.py:228]: The response.json() is {id: None}. Not sure why--as far as I can tell I'm not returning anything

File: bc_obps/registration/tests/endpoints/v2/_operations/_operation_id/_registration/test_operation_representative.py [hunk 2]
```diff
@@ -171,3 +171,73 @@ def test_register_operation_operation_representative_endpoint_bad_data(self):
 
         # Assert
         assert response.status_code == 422
+
+
+class TestOperationRepresentativePutEndpoint(CommonTestSetup):
```
Comment (BCerki) [inline bc_obps/registration/tests/endpoints/v2/_operations/_operation_id/_registration/test_operation_representative.py:176]: just wrote :musical_note: all these tests :partying_face: we don't need anymore :notes: cause Sepehr did the beautiful permissions tests :partying_face:

**Review by andrea-williams — COMMENTED** [review]
Minutiae that you shouldn't bother fixing in this PR because IMO it's entirely fine to leave these enhancements for post-MVP, but recording them here and now in hopes that someone else will remember these because I'll definitely forget:

- mousing over the garbage button shows as a pointer rather than a gloved hand as our other buttons do, so it's not as apparent that it's a button
- garbage can's background colour doesn't change on hover-over, whereas other buttons do
- I would include screenshots for these but Mac hides the pointer when you try to take a screenshot

**Comment by andrea-williams** [comment]
Notes that are a bigger deal: we don't show the operation representatives in the Operation Details page. So the workflow for external users would go like this:

1. go to Contacts tile, see "Places Assigned" for coworker Henry Ives who has now left the company. I see a note saying I must assign a different op rep for my registered operation before I can delete Henry Ives as a contact
2. LOL I can't assign a different op rep because I can't get back to the registration multistep form now that the operation has been registered, and my operation reps don't appear in the Operation Details page
3. Henry Ives must rescind his resignation letter

**Comment by BCerki** [comment]
> Notes that are a bigger deal: we don't show the operation representatives in the Operation Details page. So the workflow for external users would go like this:
> 
> 1. go to Contacts tile, see "Places Assigned" for coworker Henry Ives who has now left the company. I see a note saying I must assign a different op rep for my registered operation before I can delete Henry Ives as a contact
> 2. LOL I can't assign a different op rep because I can't get back to the registration multistep form now that the operation has been registered, and my operation reps don't appear in the Operation Details page
> 3. Henry Ives must rescind his resignation letter

Operation reps are supposed to show on the details page: https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18025-1489&t=5B28WWF1rsdwkk9u-4. I think it's covered by this ticket: https://github.com/orgs/bcgov/projects/123/views/16?filterQuery=repo%3Abcgov%2Fcas-registration+-status%3Adone++operation+rep&pane=issue&itemId=82101543&issue=bcgov%7Ccas-registration%7C2327

File: bciers/libs/components/src/form/widgets/readOnly/OperationRepresentativeWidget.test.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  type: "object",
+  properties: {
+    operationRepresentativeTestField: {
+      type: "array",
+      title: operationRepresentativeWidgetLabel,
+      minItems: 1,
+      items: {
+        type: "string",
+        enum: [1, 2],
+        enumNames: ["Neville Flashdance", "Oz Twindlewinks"],
```
Comment (andrea-williams) [inline bciers/libs/components/src/form/widgets/readOnly/OperationRepresentativeWidget.test.tsx:20]: 😂

**Review by andrea-williams — APPROVED** [review]
looking good!

Apart from the merge conflict, just a reminder to update the `endpoints_to_test` dictionary in the `TestEndpointPermissions` class :)

File: bc_obps/registration/api/v2/_operations/_operation_id/_registration/operation_representative.py [hunk 4]
```diff
@@ -28,3 +32,20 @@ def create_operation_representative(
...
+    response={200: OperationRepresentativeOut, custom_codes_4xx: Message},
+    tags=V2,
+    description="""Removes operation representative from an operation.
+    The endpoint ensures that only authorized industry users can update operations belonging to their operator. Unauthorized access attempts raise an error.""",
+    auth=authorize("approved_industry_user"),
+)
+@handle_http_errors()
+def remove_operation_representative(
+    request: HttpRequest, operation_id: UUID, payload: OperationRepresentativeRemove
+) -> Tuple[Literal[200], None]:
```
Comment (BCerki) [inline bc_obps/registration/api/v2/_operations/_operation_id/_registration/operation_representative.py:48]: @andrea-williams , is it fine to return None here? I don't think we actually need anything back. Even though it's None, what I actually get back is {'id': None}, not straight up None or {}

**Comment by BCerki** [comment]
> Minutiae that you shouldn't bother fixing in this PR because IMO it's entirely fine to leave these enhancements for post-MVP, but recording them here and now in hopes that someone else will remember these because I'll definitely forget:
> 
> * mousing over the garbage button shows as a pointer rather than a gloved hand as our other buttons do, so it's not as apparent that it's a button
> * garbage can's background colour doesn't change on hover-over, whereas other buttons do
> * I would include screenshots for these but Mac hides the pointer when you try to take a screenshot

It's not a button, it's just an icon, so I'll make it a button and that will hopefully fix. Here's my opportunity to practice CSS...

**Pull request merged**